grype/db/v6/affected_package_store.go

```diff
@@ -169,10 +178,74 @@ func (s *affectedPackageStore) getNonDistroPackageByName(packageName string, con
 	return pkgs, nil
 }
 
-func (s *affectedPackageStore) getPackageByNameAndDistro(packageName string, config GetAffectedPackageOptions) ([]AffectedPackageHandle, error) {
+func (s *affectedPackageStore) handlePackageName(query *gorm.DB, packageName string) *gorm.DB {
+	return query.Joins("JOIN packages ON affected_package_handles.package_id = packages.id").Where("packages.name = ?", packageName)
+}
+
+func (s *affectedPackageStore) handlePackageCPE(query *gorm.DB, c cpe.Attributes) *gorm.DB {
+	query = query.Joins("JOIN packages ON affected_package_handles.package_id = packages.id").Joins("JOIN cpes ON packages.id = cpes.package_id")
+
+	if c.Part != cpe.Any {
+		query = query.Where("cpes.part = ?", c.Part)
+	}
+
+	if c.Vendor != cpe.Any {
+		query = query.Where("cpes.vendor = ?", c.Vendor)
+	}
+
+	if c.Product != cpe.Any {
+		query = query.Where("cpes.product = ?", c.Product)
+	}
+
+	if c.Version != cpe.Any {
+		query = query.Where("cpes.version = ?", c.Version)
+	}
+
+	if c.Update != cpe.Any {
+		query = query.Where("cpes.update = ?", c.Update)
+	}
```

Comment:
We don't keep version and update in the CPE table since those are always just part of the affected version constraints

Reply:
good catch

Reply:
updated!